pkg/query/flamegraph_arrow.go

```diff
@@ -261,6 +264,31 @@ func generateFlamegraphArrowRecord(ctx context.Context, mem memory.Allocator, tr
 				}
 			}
 
+			sampleLabelsKey := ""
+			if _, aggregateLabels := aggregateFields[FlamegraphFieldLabels]; aggregateLabels && len(sampleLabels) > 0 {
+				lsbytes = lsbytes[:0]
+				lsbytes = MarshalStringMap(lsbytes, sampleLabels)
+				sampleLabelsKey = bytesToString(lsbytes)
+
+				sampleLabelRow := row
+				if _, ok := rootsRow[sampleLabelsKey]; ok {
+					sampleLabelRow = rootsRow[sampleLabelsKey][0]
+					compareRows = compareRows[:0] //  reset the compare rows
+					// We want to compare against this found label root's children.
+					rootRow := rootsRow[sampleLabelsKey][0]
+					compareRows = append(compareRows, fb.children[rootRow]...)
+					fb.addRowValues(r, sampleLabelRow, i) // adds the cumulative and diff values to the existing row
+				} else {
+					err := fb.AppendLabelRow(r, sampleLabelRow, sampleLabelsKey, sampleLabels, i)
+					if err != nil {
+						return nil, 0, 0, 0, fmt.Errorf("failed to inject label row: %w", err)
+					}
+					rootsRow[sampleLabelsKey] = []int{sampleLabelRow}
```

Comment:
this is unsafe, when storing something the string is kept by the map, it's only safe to use for accesses

I would recommend to just use `bytesToString` directly inline wherever there are accesses to the map are done since the "conversion" is extremely cheap and it's harder to do something wrong.